Beam Sync test is flaky

### What is wrong?

Occasionally, the beam sync test fails with a complaint like:

```
=================================== FAILURES ===================================
_____________________________ test_beam_syncer[68] _____________________________
[gw1] linux -- Python 3.7.4 /home/circleci/repo/.tox/py37-eth1-core/bin/python

request = <FixtureRequest for <Function 'test_beam_syncer[68]'>>
event_loop = <uvloop.Loop running=False closed=False debug=False>
event_bus = <networking>
chaindb_fresh = <tests.core.integration_test_helpers.FakeAsyncChainDB object at 0x7f2841b42050>
chaindb_churner = <tests.core.integration_test_helpers.FakeAsyncChainDB object at 0x7f2841c1d890>
beam_to_block = 68

    @pytest.mark.asyncio
    @pytest.mark.parametrize('beam_to_block', [1, 66, 68, 129])
    async def test_beam_syncer(
            request,
            event_loop,
            event_bus,
            chaindb_fresh,
            chaindb_churner,
            beam_to_block):
    
        client_peer, server_peer = await get_directly_linked_peers(
            request, event_loop,
            alice_headerdb=FakeAsyncHeaderDB(chaindb_fresh.db),
            bob_headerdb=FakeAsyncHeaderDB(chaindb_churner.db))
    
        # manually add endpoint for beam vm to make requests
        pausing_config = ConnectionConfig.from_name("PausingEndpoint")
    
        # manually add endpoint for trie data gatherer to serve requests
        gatherer_config = ConnectionConfig.from_name("GathererEndpoint")
    
        client_peer_pool = MockPeerPoolWithConnectedPeers([client_peer])
        server_peer_pool = MockPeerPoolWithConnectedPeers([server_peer], event_bus=event_bus)
    
        async with run_peer_pool_event_server(
            event_bus, server_peer_pool, handler_type=ETHPeerPoolEventServer
        ), run_request_server(
            event_bus, FakeAsyncChainDB(chaindb_churner.db)
        ), AsyncioEndpoint.serve(
            pausing_config
        ) as pausing_endpoint, AsyncioEndpoint.serve(gatherer_config) as gatherer_endpoint:
    
            BeamPetersburgVM = pausing_vm_decorator(PetersburgVM, pausing_endpoint, event_loop)
    
            class BeamPetersburgTestChain(FakeAsyncChain):
                vm_configuration = ((0, BeamPetersburgVM),)
                network_id = 999
    
            client_chain = BeamPetersburgTestChain(chaindb_fresh.db)
            client = BeamSyncer(
                client_chain,
                chaindb_fresh.db,
                client_chain.chaindb,
                client_peer_pool,
                gatherer_endpoint,
                beam_to_block,
            )
    
            client_peer.logger.info("%s is serving churner blocks", client_peer)
            server_peer.logger.info("%s is syncing up churner blocks", server_peer)
    
            import_server = BlockImportServer(pausing_endpoint, client_chain, token=client.cancel_token)
            asyncio.ensure_future(import_server.run())
    
            await pausing_endpoint.connect_to_endpoints(gatherer_config)
            asyncio.ensure_future(client.run())
    
            # We can sync at least 10 blocks in 1s at current speeds, (or reach the current one)
            # Trying to keep the tests short-ish. A fuller test could always set the target header
            #   to the chaindb_churner canonical head, and increase the timeout significantly
            target_block_number = min(beam_to_block + 10, 129)
            target_head = chaindb_churner.get_canonical_block_header_by_number(target_block_number)
            await wait_for_head(chaindb_fresh, target_head, sync_timeout=4)
            assert target_head.state_root in chaindb_fresh.db
    
            # first stop the import server, so it doesn't hang waiting for state data
            await import_server.cancel()
>           await client.cancel()

BeamPetersburgTestChain = <class 'test_sync.test_beam_syncer.<locals>.BeamPetersburgTestChain'>
BeamPetersburgVM = <class 'trinity.sync.beam.importer.pausing_vm_decorator.<locals>.PausingVM'>
beam_to_block = 68
chaindb_churner = <tests.core.integration_test_helpers.FakeAsyncChainDB object at 0x7f2841c1d890>
chaindb_fresh = <tests.core.integration_test_helpers.FakeAsyncChainDB object at 0x7f2841b42050>
client     = <trinity.sync.beam.chain.BeamSyncer object at 0x7f2841c72e90>
client_chain = <test_sync.test_beam_syncer.<locals>.BeamPetersburgTestChain object at 0x7f2840558c10>
client_peer = ETHPeer <Node(0xe33f170f4dcd3c216f2ce52214f167c8d438fccce36f17afb064f4104a5700322ffb0d20a52492db2c081a214658cc2fe7f8a0b4c99370b19396930926a7dcaf@10.0.0.68:40863)>
client_peer_pool = <tests.core.peer_helpers.MockPeerPoolWithConnectedPeers object at 0x7f2841c785d0>
event_bus  = <networking>
event_loop = <uvloop.Loop running=False closed=False debug=False>
gatherer_config = ConnectionConfig(name='GathererEndpoint', path=PosixPath('GathererEndpoint.ipc'))
gatherer_endpoint = <GathererEndpoint>
import_server = <trinity.sync.beam.importer.BlockImportServer object at 0x7f2841c72cd0>
pausing_config = ConnectionConfig(name='PausingEndpoint', path=PosixPath('PausingEndpoint.ipc'))
pausing_endpoint = <PausingEndpoint>
request    = <FixtureRequest for <Function 'test_beam_syncer[68]'>>
server_peer = ETHPeer <Node(0x22478d3d6ba8c34f9b9b9e0e41e0d233826afd7909d88a6357a054b84d8c76aec35972940e25d96bce84f68672c52d951424932c386821ece1f1ee46f8b1f076@10.0.0.67:41061)>
server_peer_pool = <tests.core.peer_helpers.MockPeerPoolWithConnectedPeers object at 0x7f2841b566d0>
target_block_number = 78
target_head = <eth.rlp.headers.BlockHeader object at 0x7f2841c6e710>

tests/core/p2p-proto/test_sync.py:209: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/py37-eth1-core/lib/python3.7/site-packages/async_generator/_util.py:53: in __aexit__
    await self._agen.athrow(type, value, traceback)
.tox/py37-eth1-core/lib/python3.7/site-packages/lahja/asyncio/endpoint.py:598: in serve
    yield endpoint
.tox/py37-eth1-core/lib/python3.7/site-packages/async_generator/_util.py:53: in __aexit__
    await self._agen.athrow(type, value, traceback)
.tox/py37-eth1-core/lib/python3.7/site-packages/lahja/asyncio/endpoint.py:590: in run
    self.stop()
.tox/py37-eth1-core/lib/python3.7/site-packages/lahja/asyncio/endpoint.py:571: in stop
    self.stop_server()
.tox/py37-eth1-core/lib/python3.7/site-packages/lahja/asyncio/endpoint.py:561: in stop_server
    self.ipc_path.unlink()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('PausingEndpoint.ipc')

    def unlink(self):
        """
            Remove this file or link.
            If the path is a directory, use rmdir() instead.
            """
        if self._closed:
            self._raise_closed()
>       self._accessor.unlink(self)
E       FileNotFoundError: [Errno 2] No such file or directory: 'PausingEndpoint.ipc'

self       = PosixPath('PausingEndpoint.ipc')
```

Like in: https://circleci.com/gh/ethereum/trinity/71780

### How can it be fixed

Not sure. It's plausible that it has to do with running in pytest-xdist, and we just need to create an IPC file that's unique to the xdist process.

--- a/tests/core/p2p-proto/test_sync.py
+++ b/tests/core/p2p-proto/test_sync.py
@@ -1,4 +1,5 @@
 import asyncio
+import uuid
 
 from eth.exceptions import HeaderNotFound
 from lahja import ConnectionConfig, AsyncioEndpoint
@@ -154,11 +155,15 @@ async def test_beam_syncer(
         alice_headerdb=FakeAsyncHeaderDB(chaindb_fresh.db),
         bob_headerdb=FakeAsyncHeaderDB(chaindb_churner.db))
 
+    # Need a name that will be unique per xdist-process, otherwise
+    #   lahja IPC endpoints in each process will clobber each other
+    unique_process_name = uuid.uuid4()
+
     # manually add endpoint for beam vm to make requests
-    pausing_config = ConnectionConfig.from_name("PausingEndpoint")
+    pausing_config = ConnectionConfig.from_name(f"PausingEndpoint-{unique_process_name}")
 
     # manually add endpoint for trie data gatherer to serve requests
-    gatherer_config = ConnectionConfig.from_name("GathererEndpoint")
+    gatherer_config = ConnectionConfig.from_name(f"GathererEndpoint-{unique_process_name}")
 
     client_peer_pool = MockPeerPoolWithConnectedPeers([client_peer])
     server_peer_pool = MockPeerPoolWithConnectedPeers([server_peer], event_bus=event_bus)

